tasks/actorstate/miner.go

```diff
@@ -77,17 +84,54 @@ func (m StorageMinerExtractor) Extract(ctx context.Context, a ActorInfo, emsgs [
 		return nil, xerrors.Errorf("extracting miner posts: %v", err)
 	}
 
-	return &minermodel.MinerTaskResult{
-		Posts: posts,
-
-		MinerInfoModel:           minerInfoModel,
-		LockedFundsModel:         lockedFundsModel,
-		FeeDebtModel:             feeDebtModel,
-		CurrentDeadlineInfoModel: currDeadlineModel,
-		SectorDealsModel:         sectorDealsModel,
-		SectorEventsModel:        sectorEventsModel,
-		SectorsModel:             sectorModel,
-		PreCommitsModel:          preCommitModel,
+	// if the miner actor is v1-6 persist its model to the miner_sector_infos table
+	var sectorModelV6Minus minermodel.MinerSectorInfoV1_6List
+	if a.Actor.Code.Equals(miner0.Actor{}.Code()) ||
+		a.Actor.Code.Equals(miner2.Actor{}.Code()) ||
+		a.Actor.Code.Equals(miner3.Actor{}.Code()) ||
+		a.Actor.Code.Equals(miner4.Actor{}.Code()) ||
+		a.Actor.Code.Equals(miner5.Actor{}.Code()) ||
+		a.Actor.Code.Equals(miner6.Actor{}.Code()) {
+		for _, sm := range sectorModelV7Plus {
+			sectorModelV6Minus = append(sectorModelV6Minus, &minermodel.MinerSectorInfoV1_6{
+				Height:                sm.Height,
+				MinerID:               sm.MinerID,
+				SectorID:              sm.SectorID,
+				StateRoot:             sm.StateRoot,
+				SealedCID:             sm.SealedCID,
+				ActivationEpoch:       sm.ActivationEpoch,
+				ExpirationEpoch:       sm.ExpirationEpoch,
+				DealWeight:            sm.DealWeight,
+				VerifiedDealWeight:    sm.VerifiedDealWeight,
+				InitialPledge:         sm.InitialPledge,
+				ExpectedDayReward:     sm.ExpectedDayReward,
+				ExpectedStoragePledge: sm.ExpectedStoragePledge,
+			})
+		}
+		return &model.PersistableList{
```

Comment:
This would be cleaner if you declared the `PersistableList` first with all the other persistables and then either appended `sectorModelV7Plus` or `sectorModelV6Minus` depending on the actor code.

Reply:
agreed, done.